internal/rbd/controllerserver.go

```diff
@@ -1063,14 +1075,16 @@ func (cs *ControllerServer) CreateSnapshot(
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// Update the metadata on snapshot not on the original image
-	rbdVol.RbdImageName = rbdSnap.RbdSnapName
-
 	// Set snapshot-name/snapshot-namespace/snapshotcontent-name details
 	// on RBD backend image as metadata on create
-	err = rbdVol.setSnapshotMetadata(req.GetParameters())
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+	if cs.SetMetadata {
+		// Update the metadata on snapshot not on the original image
+		rbdVol.RbdImageName = rbdSnap.RbdSnapName
+
+		err = rbdVol.setSnapshotMetadata(req.GetParameters())
```

Comment:
can we make `SetMetadata` part of the rbdImage struct, if we do so we just need if check inside the  `setSnapshotMetadata` and `setVolumeMetadata`, and changes will be minimal and easy to remove this option later also? WDYT?